Public/Src/Cache/ContentStore/Vsts/DedupReadOnlyContentSession.cs

```diff
@@ -178,6 +178,15 @@ private enum Counters
                 return pinResult;
             }
 
+            VstsBlobIdentifier blobId = ToVstsBlobIdentifier(contentHash.ToBlobIdentifier());
+            VstsDedupIdentifier dedupId = blobId.ToDedupIdentifier();
+
+            if (dedupId.AlgorithmId == Hashing.ChunkDedupIdentifier.ChunkAlgorithmId)

```

Comment:
This optimization makes sense, but I'd like to understand when it comes into play in practice. Is the situation that, if a piece of content is small, we map it's dedup content hash directly to a single chunk hash, and if it is large, we map it to a root node hash?

Reply:
Initially I thought that if the content was small, there would always be one root node pointing to a single chunk. However, I saw build failures where the optimization was used on chunks:

ArgumentException: System.ArgumentException: The given hash does not represent a NodeDedupIdentifier
   at Microsoft.VisualStudio.Services.BlobStore.Common.NodeDedupIdentifier.Validate()
   at Microsoft.VisualStudio.Services.BlobStore.Common.DedupIdentifier.CastToNodeDedupIdentifier()